cmd/apko/cli/publish.go

```diff
@@ -37,25 +40,39 @@ in a keychain.`,
 		Example: `  apko publish <config.yaml> <tag>`,
 		Args:    cobra.ExactArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return PublishCmd(cmd.Context(), args[0], args[1])
+			digest, err := PublishCmd(cmd.Context(), args[0], args[1])
+			if err != nil {
+				return err
+			}
+			if imageRefs != "" {
```

Comment:
Would it be possible to move this into `PublishCmd`?  I'd prefer to keep our indentation levels low.

Reply:
Yeah, I'll just have to plumb the variable thru.

Reply:
We could have a PublishOptions{} struct I guess, seems like the best way to do it.

Reply:
ok done on both

Reply:
Github swallowed the second comment until just now.  One trick we use is to make things like `PublishCmd` methods on a struct, and define flags as variables on fields on that struct.

e.g.

```go
type Publish struct {
  imageRefs string
}

func (p *Publish) Execute() ...


...
p := Publish{}
cmd := &cobra.Command{
		Use:   "publish",
		Args:    cobra.ExactArgs(2),
		RunE: p.Execute,
}
cmd.Flags().StringVar(&p.imageRefs, "image-refs", "", "path to file where a list of the published image references will be written")

```

Reply:
@n3wscott refactored cosign to use this a bunch, and we can define options structs and use interfaces to build up more elaborate flag surfaces.

Reply:
Lets open an issue about it and talk about doing this refactoring in v0.2!